### DISCUSSION TITLE
Changing the background color of the IPF plot according to some selected values

### DISCUSSION THREAD

**Discussion by Mustafa-Rifat** [mtex-toolbox/mtex#919] [Ask Anything]
Hello,

I have 24 grains and corresponding 24 values of Taylor Factor. What I have currently done is plotted the mean orientation of the grains in the IPF and color coded the points using the Taylor Factor values (see the attached image).
e.g. The code that I used to plot the mean orientation of grain o in the IPF plot is shown below. M(15) is grain o's Taylor Factor value:
`plotIPDF(grain_o_mori,M(15),[vector3d.X,],'MarkerSize',(10),'Marker','d','MarkerEdgeColor','black')`
![IPF plot Taylor Factor](https://user-images.githubusercontent.com/60717121/112921415-bd685880-90d8-11eb-8f16-64cf259f17cb.jpg)
However, what I want is to do is have the background of the IPF color coded in zones according to the Taylor Factor values. So the IPF will be colored (like below). All the points showing mean orientation of the grains will not be color coded as it is now, rather they will be black. Can you help me in this regard?
![107542683-6f35df00-6b96-11eb-9d56-b148ab097d68](https://user-images.githubusercontent.com/60717121/112922270-32885d80-90da-11eb-8e8b-44fb8f3cb289.png)

**Comment by ralfHielscher** [comment]
Is it true that the Taylor factor depends only on the inverse pole figure direction? I mean in general it depends on the full orientation and can be plotted in orientation space as demonstrated here: https://mtex-toolbox.github.io/TaylorModel.html#3. 
The question is now whether they Taylor factor is constant for all orientations corresponding to one point in the inverse pole figure?
Or do you want to visualize an average Taylor factor for each inverse pole figure direction?

Ralf.

**Reply by Mustafa-Rifat** [reply]
I want to visualize an average Taylor Factor for xvector let's say.
I already have the values of the taylor factor for the grains I am interested at. I also have their mean orientation.

**Reply by Mustafa-Rifat** [reply]
To be more specific, I want something like the top figure here.
![1-s2 0-S1359645420303761-gr13](https://user-images.githubusercontent.com/60717121/113028584-6f476980-9159-11eb-920e-82265b9f592e.jpg)

**Reply by King-wang-H** [reply]
Hi, I have also learned how to draw such a map recently. Have you solved the problem in this aspect?
Could you please provide a demo code for my reference?
Thank you very much.

**Comment by ralfHielscher** [comment]
Hi Rihat,

I saw that you commented on a Schmid factor post in the old MTEX forum. 
You can the following. But you need to make sure it really compute what you want.

```matlab
% set up crystal symmetry
cs = crystalSymmetry('m-3m');

% define a family of slip systems
sS = slipSystem.bcc(cs);

% define tenstion directions
d = Miller(plotS2Grid(cs.fundamentalSector),cs);

% strain rate 
q = 0.1;

% define differen strain tensors with the different tension directions
epsilon = strainRateTensor.uniaxial(d,q)

% compute the Taylor factor
[M,b,W] = calcTaylor(epsilon,sS.symmetrise);

% plot the result
plot(d,M)
```

Ralf.
